sdk/python/kfp/dsl/_component_bridge.py

```diff
@@ -507,17 +509,13 @@ def _resolve_ir_placeholders_v2(
           pipeline_task_spec.inputs.artifacts[
               input_name].task_output_artifact.output_artifact_key = (
                   argument_value.name)
-        elif is_compiling_for_v2:
-          # argument_value.op_name could be none, in which case an importer node
-          # will be inserted later.
-          # Importer node is only applicable for v2 engine.
-          pipeline_task_spec.inputs.artifacts[
-              input_name].task_output_artifact.producer_task = ''
-          type_schema = type_utils.get_input_artifact_type_schema(
-              input_name, component_spec.inputs)
-          importer_specs[input_name] = importer_node.build_importer_spec(
-              input_type_schema=type_schema,
-              pipeline_param_name=argument_value.name)
+        else:
+          # Argument is a reference to a pipeline input.
+          # Pipeline inputs can only be parameters. Fixing accordingly.
+          if not type_utils.is_parameter_type(input_type):
```

Comment:
Instead of relying on `is_parameter_type`, should we check for `input_type` is None or empty? Otherwise we could alter user's intention quietly.

Reply:
This is intentional.
An important case which the PR fixes is situation when the user passes a constant argument to an input which is typed, but not recognized as a parameter. For example, `Date`, `URI`, `InputConfig`, `PythonCode`, etc.
Same with the pipeline parameters with types like `Date`, `URI`, `InputConfig`.

The user's intention is pretty clear and unambiguous: Pass constant or pipeline parameter to a component (which consumes it by value).
It would looks strange if we start asking users to stop typing their inputs. I do not think we should as users to change `Date`, `URI`, `InputConfig`, `GcsUri` to `String`.

Reply:
> For example, Date, URI, InputConfig, PythonCode, etc.

For these types, we have another action item to whitelist them as string parameters. That way we can cover the case when the value is not present at compile time, which is the case this PR doesn't handle, and will cause inconsistency. WDYT?

Reply:
I'm not sure enumerating all possible type names in the SDK is maintainable.
It also does not cover the custom types that our users might be using.

The user wants to pass constant argument to a component. The component wants to consume an input by value.

This works in KFP and this PR provides a way to make it work in IR. Since the user intent is clear, I do not see a reason to stop the user scenarios from working.

Reply:
Right, on a second thought, whitelisting is really not a good approach.